ledger/txtail.go

```diff
@@ -115,14 +117,18 @@ func (t *txTail) loadFromDisk(l ledgerForTracker, dbRound basics.Round) error {
 	}
 
 	t.lowWaterMark = l.Latest()
-	t.lastValid = make(map[basics.Round]map[transactions.Txid]struct{})
+	t.lastValid = make(map[basics.Round]map[transactions.Txid]int16)
```

Comment:
I have often thought we could do a nice thing with these sorts of maps that hold a sequence of rounds mapped to something and operate as double-ended queues - old stuff falls off, new stuff comes on - and have a maximum size.  We could have a circular buffer and offsets.  It would have the extra advantage of being kept in order, so we could walk backward while searching for things.  Anyway, another PR someday.